src/H5system.c

```diff
@@ -935,8 +935,170 @@ H5_expand_windows_env_vars(char **env_var)
 
     FUNC_LEAVE_NOAPI(ret_value)
 } /* end H5_expand_windows_env_vars() */
+
+char *
+H5_strndup(const char *s, size_t n)
+{
+    size_t len       = 0;
+    char  *ret_value = NULL;
+
+    FUNC_ENTER_NOAPI_NOINIT
+
+    if (!s)
+        HGOTO_ERROR(H5E_ARGS, H5E_BADVALUE, NULL, "string cannot be NULL")
+
+    len = HDstrlen(s);
+
+    if (len > n)
+        len = n;
+
+    if (NULL == (ret_value = HDmalloc(len + 1)))
```

Comment:
Should use H55MM_malloc here for consistency with other routines in this file.

Reply:
Had to add a new H5MM_strndup to satisfy this for H5_dirname and H5_basename, but done.